### ISSUE TITLE
Editor jumps to end of file

### ISSUE THREAD

**Issue by harveyr** [Microsoft/vscode#47759]
Issue Type: <b>Bug</b>

In a Typescript file with several linter/compiler errors, and with format-on-save enabled:

1. Save the file, which triggers a formatting change.
1. After the save and autoformat are complete, if I hit the right-arrow key (with no other keys held), the editor jumps to the end of the file. 

This seems to be new behavior as of the last release or two. I hadn't pinned down what was happening until just now. At first I thought maybe I was hitting the arrow key too quickly, lumping it into some key chord, but this happens even if I wait several seconds after the save.

As mentioned, the file has several linter/compiler errors after the save. The problem doesn't seem to occur in other files with no errors, but I didn't do very robust testing. Here's the compiler output (I'm in the middle of some refactoring):

```
src/deploys/util.ts(199,7): error TS2451: Cannot redeclare block-scoped variable 'attachment'.
src/deploys/util.ts(205,5): error TS2304: Cannot find name 'created'.
src/deploys/util.ts(222,16): error TS2304: Cannot find name 'inputStr'.
src/deploys/util.ts(223,57): error TS2304: Cannot find name 'inputStr'.
src/deploys/util.ts(226,9): error TS2451: Cannot redeclare block-scoped variable 'attachment'.
src/deploys/util.ts(236,16): error TS2304: Cannot find name 'defaultValue'.
src/deploys/util.ts(243,3): error TS2322: Type 'MessageAttachment | undefined' is not assignable to type 'MessageAttachment'.
  Type 'undefined' is not assignable to type 'MessageAttachment'.
```

VS Code version: Code 1.22.1 (950b8b0d37a9b7061b6f0d291837ccc4015f5ecd, 2018-04-06T00:21:13.607Z)
OS version: Darwin x64 17.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2900)|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.28GB free)|
|Process Argv|/Applications/Visual Studio Code.app/Contents/MacOS/Electron -psn_0_930019|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (20)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-docker|Pet|0.0.26
Bookmarks|ale|0.18.2
cucumberautocomplete|ale|2.9.0
django-html|bib|1.2.0
better-toml|bun|0.3.2
plsql|cas|0.0.3
vscode-standardjs|che|1.2.1
gitignore|cod|0.5.0
mustache|daw|1.1.1
gitlens|eam|8.2.1
prettier-vscode|esb|1.2.2
black-vscode|jos|2.0.2
recent-files|jun|1.1.10
Go|luk|0.6.78
python|ms-|2018.3.1
jenkinsfile-support|sec|0.1.0
rewrap|stk|1.8.1
vscode-todo-highlight|way|0.5.12
jinja|who|0.0.8
vscode-open-in-github|ziy|1.3.1

(1 theme extensions excluded)

</details>

**Comment by mjbvz** [comment]
Duplicate of #47382 Should be fixed in an upcoming 1.22.2 release

**Comment by harveyr** [comment]
👍 Thanks!
